### PR TITLE
Fix bug on Windows with poll initialization for sockets

### PR DESCRIPTION
**Pull request by benh**

### CODE REVIEW

File: eventuals/event-loop.h [hunk 1]
```diff
@@ -797,7 +803,8 @@ class EventLoop final : public Scheduler {
             &context_);
       }
 
-      void Next() override {
+      void
```
Comment (onelxj) [inline eventuals/event-loop.h:806]: Restore?

**Review by onelxj — APPROVED** [review]

File: eventuals/event-loop.h [hunk 2]
```diff
@@ -781,9 +781,15 @@ class EventLoop final : public Scheduler {
               if (!completed_) {
                 CHECK(!started_);
                 started_ = true;
-
+#if !defined(_WIN32)
```
Comment (onelxj) [inline eventuals/event-loop.h:784]: Maybe `#ifndef` could be better there?

**Comment by FolMing** [comment]
Come on, guys, why approve? This doesn't work at all!

1. GetFileType requires #include <fileapi.h>
2. GetFileType expects HANDLE as a parameter and not a random integer.

Even after fixing the above things (using reinterpret_cast for the 2.) the test executable crashes:
image

File: eventuals/event-loop.h [hunk 3]
```diff
@@ -781,9 +781,15 @@ class EventLoop final : public Scheduler {
...
                 error_ = uv_poll_init(loop_, poll(), fd_);
-
+#else
+                if (GetFileType(fd_) == FILE_TYPE_PIPE) {
```
Comment (FolMing) [inline eventuals/event-loop.h:787]: 1. GetFileType requires #include <fileapi.h>
2. GetFileType expects HANDLE as a parameter and not a random integer.

**Review by FolMing — CHANGES_REQUESTED** [review]

**Comment by FolMing** [comment]
#346 should fix the issues above.

**Review by FolMing — APPROVED** [review]

**Pull request merged**